app/src/main/java/de/xikolo/controllers/downloads/DownloadsFragment.kt

```diff
@@ -0,0 +1,253 @@
+package de.xikolo.controllers.downloads
+
+import android.os.Bundle
+import android.support.v4.app.DialogFragment
+import android.support.v4.app.Fragment
+import android.support.v4.widget.SwipeRefreshLayout
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import de.xikolo.R
+import de.xikolo.config.FeatureToggle
+import de.xikolo.controllers.dialogs.ConfirmDeleteDialog
+import de.xikolo.controllers.helper.LoadingStateHelper
+import de.xikolo.events.PermissionDeniedEvent
+import de.xikolo.events.PermissionGrantedEvent
+import de.xikolo.managers.DownloadManager
+import de.xikolo.managers.PermissionManager
+import de.xikolo.storages.ApplicationPreferences
+import de.xikolo.utils.FileUtil
+import de.xikolo.utils.StorageUtil
+import de.xikolo.utils.ToastUtil
+import org.greenrobot.eventbus.EventBus
+import org.greenrobot.eventbus.Subscribe
+import java.io.File
+import java.util.*
+
+class DownloadsFragment : Fragment(), SwipeRefreshLayout.OnRefreshListener, DownloadsAdapter.OnDeleteButtonClickedListener {
+
+    companion object {
+        val TAG: String = DownloadsFragment::class.java.simpleName
+    }
+
+    private var adapter: DownloadsAdapter? = null
```

Comment:
Can't we do these as `lateinit` to avoid nullability?

Reply:
They all kinda depend on `activity`, which is marked as nullable now.